tollway/__main__.py

```diff
@@ -0,0 +1,124 @@
+import time
+import random
+import json
+from typing import Optional
+from datetime import datetime, timedelta
+from pathlib import Path
+
+import typer
+from faker import Faker
+from faker_vehicle import VehicleProvider
+
+from tollway.vehicle import (
+    get_tollways, 
+    create_tollway, 
+    create_vehicle, 
+    create_payload,
+    TIMESTAMP_FORMAT,
+)
+
+
+DATE_VARIATION_RATE = 3
+INCLUDE_LATE_RATE = 20
+INCLUDE_DUPLICATE_RATE = 50
+ALL_EVENTS_COUNT = 100
+
+tollways = get_tollways()
+fake = Faker()
+fake.add_provider(VehicleProvider)
+
+
+def _get_random_timestamp(timestamps: list[str]) -> str:
+    first_timestamp = datetime.strptime(timestamps[0], TIMESTAMP_FORMAT)
+    middle_timestamp = datetime.strptime(timestamps[len(timestamps) // 2], TIMESTAMP_FORMAT)
+
+    random_percentage = random.random()
+    timestamp_range = middle_timestamp - first_timestamp
+    random_timestamp = (first_timestamp + random_percentage * timestamp_range).strftime(TIMESTAMP_FORMAT)
+    return random_timestamp
+
+
+def _push_to_topic():
+    pass # to be developed
+
+
+def _write_to_file(filename: str, events: list[dict]):
```

Comment:
you could move these methods out (making them "public") so __main__ only contains the main() = driver of the package

Reply:
good call, I will create another module, thanks